courses/models_test.py

```diff
@@ -523,7 +523,7 @@ def test_program_certificate_start_end_dates_and_page_revision(user):
     certificate = ProgramCertificateFactory.create(program=program, user=user)
     program_start_date, program_end_date = certificate.start_end_dates
     assert program_start_date == early_course_run.start_date
-    assert program_end_date == later_course_run.end_date
+    assert program_end_date == certificate.created_on
```

Comment:
The variable `program_end_date` should probably be renamed since it's not actually equal to the program's end_date.  I would also update the docstring for this unit test - I would describe how the test ensures `start_end_dates` is working as expected.

Reply:
I think ` program_end_date` sounds correct in its meaning. Because the program end date is unique for each learner dependent on when they finished all the courses.

Reply:
Does the program model have an `end_date` attribute?